ci/lava/conftest.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""Pytest configuration file."""
+
+import pytest
+import re
+import subprocess
+import sys
+
+
+def pytest_report_teststatus(report):
+    """Override Pytest hook to report test status."""
+    if report.when == "call":
+        if report.passed:
+            lava = "<LAVA_SIGNAL_TESTCASE TEST_CASE_ID={} RESULT=pass>".format(
+                report.nodeid.replace(" ", "_")
+            )
+            return report.outcome, "*", lava
+
+    if report.failed:
+        lava = "<LAVA_SIGNAL_TESTCASE TEST_CASE_ID={} RESULT=fail>".format(
+            report.nodeid.replace(" ", "_")
+        )
+        return report.outcome, "*", lava
+
+
+def pytest_addoption(parser):
+    """Add option parsing to pytest."""
+    parser.addoption("--device", action="store", default="none")
+    parser.addoption("--dut", action="store", default="auto")
+
+
+@pytest.fixture
+def device(request):
+    """Fixture for --device."""
+    return request.config.getoption("--device")
+
+
+@pytest.fixture
+def dut(request):
+    """Fixture for --dut."""
+    return request.config.getoption("--dut")
+
+
+class Execute_Helper:
+    """Class to provide a wrapper for executing commands as a subprocess."""
+
+    @staticmethod
+    def _print(data):
+        # print(data)
+        pass
+
+    @staticmethod
+    def _execute_command(command):
+        """Execute the provided command list.
+
+        Executes the command and returns the erro code, stdout and stderr.
+        """
+        Execute_Helper._print("execute_command start:")
+        Execute_Helper._print("command:")
+        Execute_Helper._print(command)
+        p = subprocess.Popen(
+            command,
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            bufsize=-1,
+            universal_newlines=True,
+        )
+        output, error = p.communicate()
+        Execute_Helper._print("output:")
+        Execute_Helper._print(output)
+        Execute_Helper._print("error:")
+        Execute_Helper._print(error)
+        Execute_Helper._print("returnCode:")
+        Execute_Helper._print(p.returncode)
+        Execute_Helper._print("execute_command done.")
+
+        return p.returncode, output, error
+
+    @staticmethod
+    def _send_mbl_cli_command(command, addr):
+
+        err = 1
+        output = ""
+        error = ""
+
+        if addr != "":
+            cliCommand = ["mbl-cli", "--address", addr]
+            for item in command:
+                cliCommand.append(item)
+            err, output, error = Execute_Helper._execute_command(cliCommand)
+        return err, output, error
+
+
+@pytest.fixture
+def execute_helper():
+    """Fixture to return the helper class."""
+    return Execute_Helper
+
+
+@pytest.fixture
+def dut_addr(dut, execute_helper):
+    """Fixture to calculate the DUT IP address."""
```

Comment:
Add comment explaining dut parameter - "auto" and regexp options?

Reply:
Added comment.